pkgs/applications/networking/instant-messengers/ripcord/default.nix

```
@@ -0,0 +1,81 @@
+{ config, lib, stdenv, fetchurl, writeScript, runtimeShell, makeFontsConf,
+  p7zip, autoPatchelfHook, desktop-file-utils, imagemagick, makeWrapper,
+  twemoji-color-font, xorg, libsodium, libopus, libGL, zlib, qt5, alsaLib }:
+
+stdenv.mkDerivation rec {
+  pname = "ripcord";
+  version = "0.4.23";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://cancel.fm/dl/Ripcord-${version}-x86_64.AppImage";
+    sha256 = "0395w0pwr1cz8ichcbyrsscmm2p7srgjk4vkqvqgwyx41prm0x2h";
+    name = "${pname}-${version}.AppImage";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook p7zip desktop-file-utils imagemagick makeWrapper ];
+  buildInputs = [ libsodium libopus libGL alsaLib ] ++
+                (with xorg; [ libX11 libXScrnSaver libXcursor xkeyboardconfig ]) ++
+                (with qt5; [ qtbase qtsvg qtmultimedia qtwebsockets ]);
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    # avoid usage of appimage's runner option --appimage-extract
+    mkdir ${pname}-${version}
+    7z x $src -o${pname}-${version}
+
+    runHook postUnpack
+  '';
+
+  patchPhase = ''
+    runHook prePatch
+
+    # fixup desktop file
+    desktop-file-edit \
+      --set-key Exec --set-value ripcord \
+      --set-key Icon --set-value ripcord \
+      --set-key Comment --set-value "${meta.description}" \
+      ${pname}-${version}/Ripcord.desktop
+
+    runHook postPatch
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    cd ${pname}-${version}
+    cp -r . $out
+    install -Dm755 Ripcord.desktop $out/share/applications/ripcord.desktop
+
+    for size in 16 32 48 64 72 96 128 192 256 512 1024; do
+      mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
+      convert -resize "$size"x"$size" Ripcord_Icon.png $out/share/icons/hicolor/"$size"x"$size"/apps/ripcord.png
+    done
+
+    chmod 755 $out/$(cat AppRun)
+    makeWrapper $out/$(cat AppRun) $out/bin/ripcord \
```

```suggestion
    makeQtWrapper $out/$(cat AppRun) $out/bin/ripcord \
```

 you should also set `dontWrapQtApps = true;`

---

You actually have to adjust the wrapping to happen in `preFixup` for it to be populated with the correct entries.

---

Notice : since AppRun can be a shell script or a symlink, i tend to not use it. That's why i work on a desktop entry getter.